diff --git a/Wrapping/Tools/CMakeLists.txt b/Wrapping/Tools/CMakeLists.txt
index ee2377a33ff..0dfdbe805bc 100644
--- a/Wrapping/Tools/CMakeLists.txt
+++ b/Wrapping/Tools/CMakeLists.txt
@@ -50,6 +50,7 @@ set_property(SOURCE vtkParseMain.c
 set(_vtk_module_no_namespace_abi_mangling ON)
 
 vtk_module_add_module("${ns}::WrappingTools"
+  LIBRARY_NAME_SUFFIX "_CompileTools"  # distinct from port vtk
   SOURCES ${sources}
   HEADERS ${headers})
 if (CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
