commit f10b14f99bf10146b7e1912fb8778005d8904faa
Author: HanByul Yang <hanbyul.yang@raymedical.com>
Date:   Tue Sep 9 15:54:56 2025 +0900

    fix-mpi

diff --git a/cmake/TPLs/FindTPLParMETIS.cmake b/cmake/TPLs/FindTPLParMETIS.cmake
index 2dc2c149de..9bc04ae82d 100644
--- a/cmake/TPLs/FindTPLParMETIS.cmake
+++ b/cmake/TPLs/FindTPLParMETIS.cmake
@@ -54,7 +54,8 @@
 # @HEADER
 
 
-TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( ParMETIS
-  REQUIRED_HEADERS parmetis.h
-  REQUIRED_LIBS_NAMES "parmetis;metis"
-  )
+find_package(parmetis REQUIRED)
+tribits_extpkg_create_imported_all_libs_target_and_config_file(
+  ParMETIS
+  INNER_FIND_PACKAGE_NAME parmetis
+  IMPORTED_TARGETS_FOR_ALL_LIBS parmetis )
diff --git a/packages/seacas/libraries/ioss/src/Ioss_Decomposition.h b/packages/seacas/libraries/ioss/src/Ioss_Decomposition.h
index 14ca00bd4e..646a3941be 100644
--- a/packages/seacas/libraries/ioss/src/Ioss_Decomposition.h
+++ b/packages/seacas/libraries/ioss/src/Ioss_Decomposition.h
@@ -94,6 +94,8 @@ namespace Ioss {
     SetDecompositionData()                             = default;
     SetDecompositionData(const SetDecompositionData &) = delete;
     SetDecompositionData(SetDecompositionData &&)      = default;
+    SetDecompositionData &operator =(const SetDecompositionData &) = delete;
+    SetDecompositionData &operator =(SetDecompositionData &&) = default;
 
     ~SetDecompositionData()
     {
diff --git a/packages/seacas/libraries/ioss/src/exodus/Ioex_DecompositionData.h b/packages/seacas/libraries/ioss/src/exodus/Ioex_DecompositionData.h
index 5faa8dc3c2..b37836dbaf 100644
--- a/packages/seacas/libraries/ioss/src/exodus/Ioex_DecompositionData.h
+++ b/packages/seacas/libraries/ioss/src/exodus/Ioex_DecompositionData.h
@@ -46,7 +46,11 @@ namespace Ioex {
   class IOEX_EXPORT DecompositionDataBase
   {
   public:
-    DecompositionDataBase(Ioss_MPI_Comm comm) : comm_(comm) {}
+    explicit DecompositionDataBase(Ioss_MPI_Comm comm) : comm_(comm) {};
+    DecompositionDataBase(const DecompositionDataBase &) = delete;
+    DecompositionDataBase(DecompositionDataBase &&)      = delete;
+    DecompositionDataBase &operator =(const DecompositionDataBase &) = delete;
+    DecompositionDataBase &operator =(DecompositionDataBase &&) = delete;
     DecompositionDataBase(const DecompositionDataBase &)            = delete;
     DecompositionDataBase &operator=(const DecompositionDataBase &) = delete;
 
diff --git a/packages/seacas/libraries/ioss/src/exodus/Ioex_ParallelDatabaseIO.h b/packages/seacas/libraries/ioss/src/exodus/Ioex_ParallelDatabaseIO.h
index 2ab679de54..28ca56e0e5 100644
--- a/packages/seacas/libraries/ioss/src/exodus/Ioex_ParallelDatabaseIO.h
+++ b/packages/seacas/libraries/ioss/src/exodus/Ioex_ParallelDatabaseIO.h
@@ -16,6 +16,7 @@
 #include "Ioss_Map.h"                   // for Map
 #include "Ioss_State.h"                 // for State
 #include "exodus/Ioex_BaseDatabaseIO.h" // for DatabaseIO
+#include "exodus/Ioex_DecompositionData.h"
 #include <functional>                   // for less
 #include <map>                          // for map, map<>::value_compare
 #include <memory>
