diff --git a/packages/seacas/libraries/ioss/src/Ioss_Field.C b/packages/seacas/libraries/ioss/src/Ioss_Field.C
index c110b825c5..0e0b33a8be 100644
--- a/packages/seacas/libraries/ioss/src/Ioss_Field.C
+++ b/packages/seacas/libraries/ioss/src/Ioss_Field.C
@@ -11,6 +11,7 @@
 #include <cstddef>
 #include <cstdint>
 #include <fmt/ostream.h>
+#include <fmt/ranges.h>
 #include <iostream>
 #include <string>
 #include <vector>
diff --git a/packages/seacas/libraries/ioss/src/Ioss_VariableType.C b/packages/seacas/libraries/ioss/src/Ioss_VariableType.C
index c14a7cfa15..21a21216f5 100644
--- a/packages/seacas/libraries/ioss/src/Ioss_VariableType.C
+++ b/packages/seacas/libraries/ioss/src/Ioss_VariableType.C
@@ -18,6 +18,7 @@
 #include <fmt/core.h>
 #include <fmt/format.h>
 #include <fmt/ostream.h>
+#include <fmt/ranges.h>
 #include <map>
 #include <sstream>
 #include <string>
diff --git a/packages/seacas/libraries/ioss/src/cgns/Iocgns_Utils.C b/packages/seacas/libraries/ioss/src/cgns/Iocgns_Utils.C
index f4fa782a21..8c514f51df 100644
--- a/packages/seacas/libraries/ioss/src/cgns/Iocgns_Utils.C
+++ b/packages/seacas/libraries/ioss/src/cgns/Iocgns_Utils.C
@@ -1088,9 +1088,9 @@ size_t Iocgns::Utils::common_write_metadata(int file_ptr, const Ioss::Region &re
   CGERR(cg_base_write(file_ptr, "Base", phys_dimension, phys_dimension, &base));
 
   CGERR(cg_goto(file_ptr, base, "end"));
-  std::time_t t    = std::time(nullptr);
-  std::string date = fmt::format("{:%Y/%m/%d}", fmt::localtime(t));
-  std::string time = fmt::format("{:%H:%M:%S}", fmt::localtime(t));
+  auto now = std::chrono::system_clock::now();
+  std::string date = fmt::format("{:%Y/%m/%d}", now);
+  std::string time = fmt::format("{:%H:%M:%S}", now);
 
   std::string code_version = region.get_optional_property("code_version", "unknown");
   std::string code_name    = region.get_optional_property("code_name", "unknown");
