diff --git a/Common/Core/vtkThreadedCallbackQueue.txx b/Common/Core/vtkThreadedCallbackQueue.txx
index b212e0adb79..71e8957f4a3 100644
--- a/Common/Core/vtkThreadedCallbackQueue.txx
+++ b/Common/Core/vtkThreadedCallbackQueue.txx
@@ -275,7 +275,7 @@ template <class T>
 struct vtkThreadedCallbackQueue::InvokerImpl::DereferenceImpl<T,
   std::true_type /* CanBeDereferencedT */>
 {
-  using Type = decltype(*std::declval<T>());
+  using Type = typename std::remove_pointer<decltype(*std::declval<T>())>::type;
   static Type& Get(T& instance) { return *instance; }
 };
 
