diff --git a/CMakeLists.txt b/CMakeLists.txt
index e9663f0d61e..a70362f5d62 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -501,7 +501,6 @@ endif ()
 # Handle the requirement of `sqlitebin` for `proj`.
 set(vtk_sqlite_build_binary 0)
 if ("VTK::libproj" IN_LIST vtk_modules)
-  set(vtk_sqlite_build_binary 1)
 endif ()
 
 cmake_dependent_option(VTK_ENABLE_KITS "Enable kits compilation" OFF
diff --git a/ThirdParty/libproj/CMakeLists.txt b/ThirdParty/libproj/CMakeLists.txt
index cea568e1232..05f85fbdb96 100644
--- a/ThirdParty/libproj/CMakeLists.txt
+++ b/ThirdParty/libproj/CMakeLists.txt
@@ -12,10 +12,12 @@ vtk_module_third_party(
       "8.1.0"
     STANDARD_INCLUDE_DIRS
   EXTERNAL
-    PACKAGE LibPROJ
-    TARGETS LibPROJ::LibPROJ
-    USE_VARIABLES LibPROJ_MAJOR_VERSION
+    PACKAGE PROJ
+    CONFIG_MODE
+    TARGETS PROJ::proj
+    USE_VARIABLES PROJ_VERSION_MAJOR
     STANDARD_INCLUDE_DIRS)
+set(LibPROJ_MAJOR_VERSION "${PROJ_VERSION_MAJOR}")
 
 if (NOT VTK_MODULE_USE_EXTERNAL_VTK_libproj)
   set(LibPROJ_MAJOR_VERSION "8")
diff --git a/ThirdParty/libproj/vtk.module b/ThirdParty/libproj/vtk.module
index 1f913cd290f..c3fe44c5526 100644
--- a/ThirdParty/libproj/vtk.module
+++ b/ThirdParty/libproj/vtk.module
@@ -6,10 +6,6 @@ SPDX_LICENSE_IDENTIFIER
   BSD-3-Clause
 SPDX_COPYRIGHT_TEXT
   Copyright (c) Ken Martin, Will Schroeder, Bill Lorensen
-DEPENDS
-  VTK::nlohmannjson
-PRIVATE_DEPENDS
-  VTK::sqlite
 TEST_DEPENDS
   VTK::TestingCore
 THIRD_PARTY
