diff --git a/ThirdParty/ioss/CMakeLists.txt b/ThirdParty/ioss/CMakeLists.txt
index 38af170ae02..af9cf68a688 100644
--- a/ThirdParty/ioss/CMakeLists.txt
+++ b/ThirdParty/ioss/CMakeLists.txt
@@ -13,7 +13,7 @@ vtk_module_third_party(
     STANDARD_INCLUDE_DIRS
   EXTERNAL
     PACKAGE SEACASIoss
-    TARGETS Ioss
+    TARGETS Ioss Ionit
     USE_VARIABLES SEACASIoss_INCLUDE_DIRS
     STANDARD_INCLUDE_DIRS)
 
